File: book/algorithm_book/typescript/unit07.ts
function selSort(a: number[]) {
  const n = a.length;

  for (let i = 0; i < n - 1; i++) {
    let min_idx = i;
    for (let j = i + 1; j < n; j++) {
      if (a[j] < a[min_idx]) {
        min_idx = j;
      }
    }
    [a[i], a[min_idx]] = [a[min_idx], a[i]];
  }

  return a;
}

let d = [2, 4, 5, 1, 3];
console.log(selSort(d));

File: book/algorithm_book/rust/src/unit_07.rs
pub fn sel_sort(a: &mut [i32]) -> &mut [i32] {
    let n = a.len();

    for i in 0..n-1 {
        let mut min_idx = i;
        for j in i+1..n {
            if a[j] < a[min_idx] {
                min_idx = j;
            }
        };
        a.swap(i, min_idx)
    }
    return a
}